plugins/eval.lua

```diff
@@ -29,4 +31,23 @@ command.add("core.docview", {
       end
     })
   end,
+
+  ["eval:selected"] = function(dv)
+    if dv.doc:has_selection() then
+      local text = dv.doc:get_text(dv.doc:get_selection())
+      dv.doc:text_input(eval(text))
+    else
+      local line = dv.doc:get_selection()
+      local text = dv.doc.lines[line]
+      dv.doc:insert(line+1, 0, "= " .. eval(text) .. "\n")
+    end
+  end,
 })
+
+
+contextmenu:register("core.docview", {
```

Comment:
Not sure if the predicate should be turned to a function that besides checking if current view is a `DocView` also checks that the current doc file extension ends in `.lua`, since we can't eval anything besides lua code and math expressions compatible with lua.

But if I recall correctly context menu ignores the given predicate and uses that of the assigned command, so this predicate could even be nil and it would use the one for the command above.

Reply:
Because of the math expressions, which very similar in many languages (and even useful in .txt files sometimes), I don't think we should restrict on the doc file extension anyway.